src/components/Modal/DeleteJobModal.js

```diff
@@ -8,11 +8,17 @@ import {
     ButtonStrip,
 } from '@dhis2/ui'
 import i18n from '@dhis2/d2-i18n'
+import { useDataMutation } from '@dhis2/app-runtime'
 import { RefetchJobsContext } from '../Context'
-import { useDeleteJob } from '../../hooks/jobs'
+
+const mutation = {
+    resource: 'jobConfigurations',
+    id: /* istanbul ignore next */ ({ id }) => id,
+    type: 'delete',
+}
 
 const DeleteJobModal = ({ id, hideModal }) => {
-    const [deleteJob] = useDeleteJob()
+    const [deleteJob] = useDataMutation(mutation)
```

Comment:
Maybe also extract out `error` and `loading`, so that we can show errors or disable the primary action button while we're loading. This comment is true for many files. Not sure what your approach to error handling is for this app, or if you intentionally left that for later.

Reply:
Yeah you're right, and it's true that it happens in several areas. I do still have to take care of that. Partially I'm also thinking it'd be good to talk about those areas of the app with Joe. Every area of the app should show feedback whenever it's doing something (like fetching, etc.), or has an error.

Reply:
I'm thinking to care of this separately (separate from this PR). What do you think?

Reply:
That sounds like a good idea. Separate PR seems appropriate.